src/Heading.php

```diff
@@ -12,7 +12,7 @@ class Heading extends Text
 
     private $level;
 
-    public function __construct($text = "", $level = self::MIN_LEVEL)
+    public function __construct($text, $level = self::MIN_LEVEL)
     {
         parent::__construct($text);
```

Comment:
Just trying to understand the syntax here. Does this mean that the validation checks on `$text`are being run on the parent 'Text' class instead of in Heading?

Reply:
@nigelhanlon yep, the parent constructor `Text::__construct` should handle the validation and assignment of this parameter.